cdap-data-fabric/src/main/java/co/cask/cdap/data2/metadata/store/DefaultMetadataStore.java

```diff
@@ -97,11 +98,16 @@
   private final TransactionExecutorFactory txExecutorFactory;
   private final DatasetFramework dsFramework;
   private AuditPublisher auditPublisher;
+  private boolean hasV1SystemDs;
```

Comment:
Missed this. These two variables should be `transient` as I believe MetadataStore could be shared from multiple threads.